### PR TITLE
Handle .NET Runtime env var overrides

### PR DESCRIPTION
**Pull request by pellared**
## Why

See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1297#discussion_r991124066

## What

Handle a case when one of `DOTNET_ADDITIONAL_DEPS`, `DOTNET_SHARED_STORE`, `DOTNET_STARTUP_HOOKS` is already set.

## Tests

Manual

## Checklist

- [x] ~~`CHANGELOG.md` is updated.~~
- [x] ~~Documentation is updated.~~
- [x] ~~New features are covered by tests.~~

### CODE REVIEW

**Comment by pellared** [comment]
I was considering making a Bash function. I think it is not worth it as it would require more Bash knowledge of the reviewers+maintainers. 

Reference: https://linuxize.com/post/bash-functions/

**Review by Kielek — APPROVED** [review]

File: instrument.sh [hunk 1]
```diff
@@ -50,13 +50,34 @@ if [ -z "$OTEL_DOTNET_AUTO_HOME" ]; then
...
+if [ -z "$DOTNET_SHARED_STORE" ]; then
+  export DOTNET_SHARED_STORE="$OTEL_DOTNET_AUTO_HOME/store"
+else
+  export DOTNET_SHARED_STORE="${DOTNET_SHARED_STORE}${SEPARATOR}${OTEL_DOTNET_AUTO_HOME}/store"
+fi
+
+if [ -z "$DOTNET_STARTUP_HOOKS" ]; then
+  export DOTNET_STARTUP_HOOKS="$OTEL_DOTNET_AUTO_HOME/netcoreapp3.1/OpenTelemetry.AutoInstrumentation.StartupHook.dll"
+else
+  export DOTNET_STARTUP_HOOKS="${DOTNET_STARTUP_HOOKS}${SEPARATOR}${OTEL_DOTNET_AUTO_HOME}/netcoreapp3.1/OpenTelemetry.AutoInstrumentation.StartupHook.dll"
```
Comment (rajkumar-rangaraj) [inline instrument.sh:79]: `DOTNET_STARTUP_HOOKS` are executed in order. Should we keep `OpenTelemetry.AutoInstrumentation.StartupHook.dll` followed by other startup hooks set?

Comment (pellared) [reply instrument.sh:79]: I think you are right. I will update the PR accordingly before merging.

**Review by pjanotti — APPROVED** [review]
Do not merge until @pellared update the PR.

File: instrument.sh (hunk 1 above)
Comment (pellared) [reply instrument.sh]: Addressed https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1390/commits/5cf294200a570a521d8eff8ad1796b75173f941e

**Pull request merged**